# Homework 3

## Code

In [1]:
import numpy as np
import scipy.stats as stats

In [ ]:
# Set data for catalysts A and B
data_A = [82, 79, 84, 78, 83, 76, 85, 80, 77, 83]
data_B = [74, 82, 80, 80, 77, 80, 78, 79, 79, 78]

# Given assumptions
mu_A = 80.7
sigma_A = 3.0
alpha = 0.05
n_B = len(data_B)
n_A = len(data_A)

# Calculate sample means
mean_A = np.mean(data_A)
mean_B = np.mean(data_B)

# Calculate standard deviations, set ddof=1 for sample standard deviation
std_A = np.std(data_A, ddof=1)
std_B = np.std(data_B, ddof=1)

# The hypothesis we need to test is whether catalyst B has a lower yield than catalyst A (mu_B < 80.7)
# This is a left-tailed test

# Test 1: One-sample Gauss test
# - we assume the true mean and true standard deviation are known
# - we also assume sigma_B = sigma_A = 3.0 for the Gauss test
Z_stat = (mean_B - mu_A) / (sigma_A / np.sqrt(n_B))
p_val_gauss = stats.norm.cdf(Z_stat)

# Test 2: One-sample one-sided t-test
# Assume the true mean of A is known exactly (mu_A = 80.7), but the standard deviation of B is estimated from the sample B.
T_stat_1 = (mean_B - mu_A) / (std_B / np.sqrt(n_B))
df_1 = n_B - 1 # Degrees of freedom: n - 1
p_val_t1 = stats.t.cdf(T_stat_1, df=df_1)

# Test 3: Two-sample one-sided t-test
# Assume both means and standard deviations are estimated from the samples.
# Assuming equal standard errors calculated from both samples.
s_p2 = ((n_A - 1) * std_A**2 + (n_B - 1) * std_B**2) / (n_A + n_B - 2)
SE   = np.sqrt(s_p2 * (1/n_A + 1/n_B))
T_stat_2 = (mean_B - mean_A) / SE
df_2     = n_A + n_B - 2
p_val_t2 = stats.t.cdf(T_stat_2, df=df_2)

# Print results
print(f"Sample B Mean: {mean_B:.3f}, Std: {std_B:.3f}")
print(f"Sample A Mean: {mean_A:.3f}, Std: {std_A:.3f}\n")
print(f"Test 1 (Gauss)          : Z = {Z_stat:.4f}, p-value = {p_val_gauss:.4f}")
print(f"Test 2 (1-Sample t-test): T = {T_stat_1:.4f}, p-value = {p_val_t1:.4f}")
print(f"Test 3 (2-Sample t-test): T = {T_stat_2:.4f}, p-value = {p_val_t2:.4f}")

Sample B Mean: 78.700, Std: 2.163
Sample A Mean: 80.700, Std: 3.129

Test 1 (Gauss)          : Z = -2.1082, p-value = 0.0175
Test 2 (1-Sample t-test): T = -2.9242, p-value = 0.0085
Test 3 (2-Sample t-test): T = -1.6628, p-value = 0.0568


## Discussion

- For test 1, the sample mean of B $(78.7)$ lies about $2.1$ standard errors $(Z)$ below $\mu_A = 80.7$. The resulting p-value $\approx 0.018$ is smaller than $\alpha = 0.05$, so we reject $H_0$ and conclude that catalyst B has a significantly lower yield than A, under the strong assumption that $\sigma = 3.0$ is the true population standard deviation.
- For test 2, the sample standard deviation $s_B \approx 2.16$ is noticeably smaller than the assumed $\sigma = 3.0$, which inflates the test statistic to $T \approx -2.92$. The p-value $\approx 0.0085$ is even smaller than in Test 1, so we again reject $H_0$, supporting the conclusion that B's yield is significantly below the reference value of $80.7$.
- For test 3, with both means estimated from finite samples we now also carry uncertainty about $\mu_A$. The pooled standard error becomes larger, $T$ drops to $\approx 1.66$, and the p-value $\approx 0.057$ is slightly greater than $\alpha = 0.05$. We therefore fail to reject $H_0$, indicating the current data do not provide quite enough evidence to conclude that B has a lower yield than A.

This mirrors the overall pattern from the lecture (slide 20): dropping assumptions introduces additional uncertainty, which typically increases the p-value. With strong assumptions (Test 1) the result is clearly significant. Once we let the data speak for themselves by estimating both means and the standard deviation (Test 3), the combined uncertainty grows and the same observation is no longer significant at the 5 % level. The qualitative impression, that B yields lower than A, is the same across all three tests, but whether the difference is statistically significant depends on how much we assume.